File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int ret = 0;
        int cnt = 0;
        for(int i = 0; i < s.size(); i++){
            if(cnt >= 0 && s[i] == '('){
                cnt++;
            }
            else if(cnt > 0 && s[i] == ')'){
                cnt--;
            }
            else if(cnt == 0 && s[i] == ')'){
                ret++;
            }
        }
        return cnt + ret;
    }
};

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int cnt = 0, ret = 0;
        char curr = ')';
        for (char c : s.toCharArray()) {
            if (curr == c && cnt == 0) ret++;
            else cnt += (c == '(') ? 1 : -1;
            curr = c;
        }
        return ((cnt > 0) ? cnt : cnt * -1) + ret;
    }
}